Modules/Tests/YosemiteTests/Tools/POS/POSCatalogSyncCoordinatorTests.swift

```diff
@@ -743,3 +582,230 @@ final class MockPOSCatalogFullSyncService: POSCatalogFullSyncServiceProtocol {
         shouldBlockSync = false
     }
 }
+
+// MARK: - Sync Eligibility Tests
+
+extension POSCatalogSyncCoordinatorTests {
+    @Test func performSmartSync_skips_sync_when_catalog_is_ineligible() async throws {
+        // Given
+        let eligibilityChecker = MockPOSLocalCatalogEligibilityService()
+        await eligibilityChecker.setIneligible(for: sampleSiteID)  // Catalog ineligible
+        let coordinator = POSCatalogSyncCoordinator(
+            fullSyncService: mockSyncService,
+            incrementalSyncService: mockIncrementalSyncService,
+            grdbManager: grdbManager,
+            catalogEligibilityChecker: eligibilityChecker,
+            siteSettings: mockSiteSettings
+        )
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: nil)
+
+        // When
+        try await coordinator.performSmartSync(for: sampleSiteID)
+
+        // Then - sync should be skipped
+        #expect(mockSyncService.startFullSyncCallCount == 0)
+        #expect(mockIncrementalSyncService.startIncrementalSyncCallCount == 0)
+    }
+
+    @Test func performSmartSync_proceeds_when_catalog_is_eligible_and_no_first_sync_date() async throws {
+        // Given - new user, catalog eligible
+        let coordinator = POSCatalogSyncCoordinator(
+            fullSyncService: mockSyncService,
+            incrementalSyncService: mockIncrementalSyncService,
+            grdbManager: grdbManager,
+            catalogEligibilityChecker: MockPOSLocalCatalogEligibilityService(), // Catalog eligible
+            siteSettings: mockSiteSettings
+        )
```

Comment:
IMHO we could be more explicit with the conditions when setting the test here, as we're testing for eligibility:

```suggestion
        // Given - new user, catalog eligible
        let eligibilityChecker = MockPOSLocalCatalogEligibilityService()
        await eligibilityChecker.setEligibility(.eligible, for: sampleSiteID)
        
        let coordinator = POSCatalogSyncCoordinator(
            fullSyncService: mockSyncService,
            incrementalSyncService: mockIncrementalSyncService,
            grdbManager: grdbManager,
            catalogEligibilityChecker: catalogEligibilityChecker,
            siteSettings: mockSiteSettings
        )
```